examples/scripts/blender/sdf_exporter.py

```diff
@@ -0,0 +1,206 @@
+        
+        if blender_light.type == "POINT":
+            light = ET.SubElement(link, "light", attrib={"name":l.name, "type":"point"})
+            diffuse = ET.SubElement(light, "diffuse")
+            diffuse.text = str(blender_light.color.r) + " " + str(blender_light.color.g) + " " + str(blender_light.color.b) + " 1.0"
+            blender_pointlight = bpy.types.PointLight(blender_light)
+            
+            add_attenuation_tags(light, blender_pointlight)
+            
+        if blender_light.type == "SPOT":
+            light = ET.SubElement(link, "light", attrib={"name":l.name, "type":"spot"})
+            diffuse = ET.SubElement(light, "diffuse")
+            diffuse.text = str(blender_light.color.r) + " " + str(blender_light.color.g) + " " + str(blender_light.color.b) + " 1.0"
+            blender_spotlight = bpy.types.SpotLight(blender_light)
+            
+            add_attenuation_tags(light, blender_spotlight)
+            # note: unsupported <spot> tags in blender
+            
+        if blender_light.type == "SUN":
+            light = ET.SubElement(link, "light", attrib={"name":l.name, "type":"directional"})
+            diffuse = ET.SubElement(light, "diffuse")
+            diffuse.text = str(blender_light.color.r) + " " + str(blender_light.color.g) + " " + str(blender_light.color.b) + " 1.0"
+            blender_pointlight = bpy.types.SunLight(blender_light)
+            
+        if blender_light.type == "SUN" or blender_light.type == "SPOT":
+            direction = ET.SubElement(light, "direction")
+            direction.text = str(l.matrix_world[0][2]) + " " + str(l.matrix_world[1][2]) + " " + str(l.matrix_world[2][2])
+            
+        # unsupported: AREA lights
+
+        cast_shadows = ET.SubElement(light, "cast_shadows")
+        cast_shadows.text = "true"
+        
+        # todo : bpy.types.light script api lacks an intensity value, possible candidate is energy/power(Watts)?
+        intensity = ET.SubElement(light, "intensity")
+        intensity.text = "1.0"
+    
+    ## sdf collision tags
+    collision = ET.SubElement(link, "collision", attrib={"name":"collision"})
+
+    geometry = ET.SubElement(collision, "geometry")
+    mesh = ET.SubElement(geometry, "mesh")
+    uri = ET.SubElement(mesh, "uri")
+    uri.text = dae_filename
+
+    surface = ET.SubElement(collision, "surface")
+    contact = ET.SubElement(collision, "contact")
+    collide_bitmask = ET.SubElement(collision, "collide_bitmask")
+    collide_bitmask.text = "0x01"
+
+    ## sdf write to file
+    xml_string = ET.tostring(sdf, encoding='unicode')
+    reparsed = minidom.parseString(xml_string)
+
+    sdf_file = open(prefix_path+sdf_filename, "w")
+    sdf_file.write(reparsed.toprettyxml(indent="  "))
+    sdf_file.close()
+
+    ##############################
+    ### generate model.config ####
+    ##############################
+    model = ET.Element('model')
+    name = ET.SubElement(model, 'name')
+    name.text = "L1"
```

Comment:
Is there a way to get this string from within Blender? Otherwise, maybe we can call it something generic like `my_model`?

Reply:
switch to `my_model`, float to top of function 3aaabae